keras/backend/jax/distribution_lib.py

```diff
@@ -40,7 +67,13 @@ def distribute_value(value, tensor_layout):
     """
     if not isinstance(tensor_layout, jax.sharding.Sharding):
         tensor_layout = _to_jax_layout(tensor_layout)
-    return jax.device_put(value, tensor_layout)
+
+    # TODO(scottzhu): This might not be a cheap check, we should consider
+    # have some proper JAX API for doing this check.
```

Comment:
I consulted mattjj and that is not something they are considering.

Reply:
Ack. Thanks for the confirmation.